### PR TITLE
docs: add coding guidelines to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by braddialpad**
Coding guidelines section

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -6,9 +6,81 @@ Thanks for your interest in contributing to Dialtone. Please take a moment to re
...
+1. Run `npm link` in your local dialtone repository root
+2. Run `npm run build` in your local dialtone repository root
+3. Run `npm link @dialpad/dialtone` in your local dialtone-vue repository root
+4. Run dialtone-vue's local storybook server to test the changes
+
+This will make your local dialtone-vue project reference your local build of dialtone rather than the dialtone dependency on npm. Note that you must re-run `npm run build` on dialtone whenever you make a change, and also if you run `npm install` in dialtone-vue it will overwrite the linked dependency to reference the npm dependency again.
+
+### A11y Standards
+
+Dialtone adopts the [WCAG 2.1 AA](https://www.w3.org/WAI/standards-guidelines/wcag/glance/) Web Content Accessibility Guidelines.
```
Comment (braddialpad) [inline .github/CONTRIBUTING.md:148]: Drew I know you are already working on a big doc for accessibility standards, so I will just link it here once it's ready.

**Review by modialpad — APPROVED** [review]

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -6,9 +6,81 @@ Thanks for your interest in contributing to Dialtone. Please take a moment to re
...
 
-## Code style
+## Coding Guidelines
 
-Make sure your code lints with `npm run lint`. If you need to format your code, you can do this with `npm run lint:fix`.
+### Immutable Utility Classes
+All of our utility classes are set to `!important`. This is because they are designed to be immutable, and `!important` is the best way we have of achieving immutability in css.
+
+### Internal classes
+While most of the classes you find in `lib/build/less` are available to users and documented on the doc site, one exception to this is d-internals. d-internals definitions are found in [utilties/internals.less](../lib/build/less/utilities/internals.less). These are definitons we use within dialtone to generate classes, and are not meant for use by consumers of dialtone.
```
Comment (dialpad-drew) [inline .github/CONTRIBUTING.md]: Is the term "internal classes" generally understood? Do you think "generated classes" may be more on-the-nose?

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -6,9 +6,81 @@ Thanks for your interest in contributing to Dialtone. Please take a moment to re
...
+
+### Naming Conventions
+All dialtone classes available to users are prefixed with `d-` example: `d-input`, `d-stack8`. Note that variables are not classes, and do not follow this naming convention. Examples: `@purple-300`, `@fw-normal`.
+
+#### Utility Classes
+Utility classes are named in the following format `d-[PROPERTY_ACRONYM]-[VALUE]`.
+
+Examples: `d-ai-center` (align-items) `d-c-pointer` (cursor)
+
+#### Components
```
Comment (dialpad-drew) [inline .github/CONTRIBUTING.md]: Is this section more specifically about "component classes?"

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -6,9 +6,81 @@ Thanks for your interest in contributing to Dialtone. Please take a moment to re
...
+Components are named using the [BEM (Block Element Modifier)](http://getbem.com/naming/) methodology. Use '__' to specify a sub-element and '--' to specify the modification of a specific property.
```
Comment (dialpad-drew) [inline .github/CONTRIBUTING.md]: Could we tighten this up and have a cleaner examples? Off-the-cuff example

"Component class names use the Block Element Modifier (BEM) naming standard."

```css
.d-banner {}  // Block
.d-banner__dialog {}  // Element
.d-banner--success {}  // Modifier
```

**Review by dialpad-drew — COMMENTED** [review]
General observations
* Make sure headings and subheadings use consistent casing (either Title Case or Sentence case).
* Make sure Dialtone is capitalized as it's a proper noun.
* I think there are a lot of really small grammatical and voice improvements that could be made here. Nothing major, but now that you've had a couple days away from it, Brad, I'd recommend re-reading it and seeing how it flows. I think we can reduce the verbosity just a hair.  (If you're having a hard time finding areas for improvement, LMK and I can go through more granularly and recommend specific things)
* Recommend trying to find a more standardized example format. What do you think about code blocks with inline comments (see feedback in Components section).

**Comment by braddialpad** [comment]
Alright, cleaned up and cut down. Let me know what you think now.

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -52,15 +52,106 @@ After you have discussed your change with the Dialtone team, follow these steps
...
 Once your change is in `staging` it will go live with the next Dialtone release. Releases are done on demand by the Dialtone team, and are done fairly regularly. If you need your change to be released promptly, please ask in the #dialtone slack channel.
 
-## Code style
+## Coding Guidelines
 
-Make sure your code lints with `npm run lint`. If you need to format your code, you can do this with `npm run lint:fix`.
+### Naming Conventions
 
-## Commit message convention
+All Dialtone classes available to users are prefixed with `d-`
```
Comment (dialpad-drew) [inline .github/CONTRIBUTING.md]: Period at the end.

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -52,15 +52,106 @@ After you have discussed your change with the Dialtone team, follow these steps
...
+
+```less
+.d-input {}
+.d-stack8 {}
+```
+
+Note that variables are not classes, and do not follow this naming convention. `@purple-300` for example.
```
Comment (dialpad-drew) [inline .github/CONTRIBUTING.md]: Ticky tack:
> Note that variables are not classes and do not follow this convention. For example: `@purple-300` or `--primary-color`.

File: .github/CONTRIBUTING.md [hunk 7]
```diff
@@ -52,15 +52,106 @@ After you have discussed your change with the Dialtone team, follow these steps
...
+
+### Immutable Utility Classes
+
+All of our utility classes are set to `!important`. This is because they are designed to be immutable, and `!important` is the best way we have of achieving immutability in CSS.
+
+### CSS Vars
+
+We use CSS vars in our classes for better reusability.
+
+Here we set the `--avatar--size` CSS var to 12, and set both the width and the height to reference this variable.
```
Comment (dialpad-drew) [inline .github/CONTRIBUTING.md]: * `12`

File: .github/CONTRIBUTING.md [hunk 8]
```diff
@@ -52,15 +52,106 @@ After you have discussed your change with the Dialtone team, follow these steps
...
+
+While most of the classes you find in `lib/build/less` are available to users and documented on the doc site, one exception to this is d-internals. d-internals definitions are found in [utilties/internals.less](../lib/build/less/utilities/internals.less). These are definitons we use within Dialtone to generate classes, and are not meant for use by consumers of Dialtone.
+
+### Immutable Utility Classes
+
+All of our utility classes are set to `!important`. This is because they are designed to be immutable, and `!important` is the best way we have of achieving immutability in CSS.
+
+### CSS Vars
+
+We use CSS vars in our classes for better reusability.
```
Comment (dialpad-drew) [inline .github/CONTRIBUTING.md]: Do you think it's worth saying that we _don't_ use Less variables? Or if we do, _when_ we do?

File: .github/CONTRIBUTING.md [hunk 9]
```diff
@@ -52,15 +52,106 @@ After you have discussed your change with the Dialtone team, follow these steps
...
+
+Here are some important directories to know within the Dialtone repository
+
+- `lib/build`: All source code and assets for the Dialtone library.
+- `lib/dist`: The compiled bundle will be output here upon `npm run build`.
+- `lib/build/fonts`: Fonts we wish to bundle with Dialtone (woff2 format).
+- `lib/build/less`: Less files defining our styles. they are processed and transpiled to CSS on build.
+- `lib/build/svg`: SVGs used for icons and spot illustrations. See [adding icons](https://dialpad.design/design/icons/adding/) for instructions on how to add icons into Dialtone.
+
+For folder structure of the doc site see the [Eleventy section](#eleventy) of this document.
```
Comment (dialpad-drew) [inline .github/CONTRIBUTING.md]: Needs comma:

> For folder structure of the doc site, ...

File: .github/CONTRIBUTING.md [hunk 10]
```diff
@@ -52,15 +52,106 @@ After you have discussed your change with the Dialtone team, follow these steps
...
+Note that we use lesshint only for the utilities folder and stylelint for everything else. Due to some unsupported syntax limitations we cannot use stylelint on our utilities folder.
+
+### Folder Structure
+
+Here are some important directories to know within the Dialtone repository
+
+- `lib/build`: All source code and assets for the Dialtone library.
+- `lib/dist`: The compiled bundle will be output here upon `npm run build`.
+- `lib/build/fonts`: Fonts we wish to bundle with Dialtone (woff2 format).
+- `lib/build/less`: Less files defining our styles. they are processed and transpiled to CSS on build.
```
Comment (dialpad-drew) [inline .github/CONTRIBUTING.md]: Is it "Less" or "LESS?"

**Review by dialpad-drew — APPROVED** [review]
@braddialpad really nice work. You're a natural writer, man. ✍️  My feedback here is super small. 🚀

**Comment by dialpad-drew** [comment]
@modialpad I know you already approved this, but if you have a few free minutes, it'd be great if you gave it another once-over since Brad did cut down on some of the length. No biggie if you don't have time, though. Appreciate you.

**Pull request merged**